### ISSUE TITLE
Not working on MacOs

### ISSUE THREAD

**Issue by FoxArtM** [alessandrofrancesconi/gimp-plugin-bimp#405]
Hi Alessandro! You've created such an amazing plugin! Thank you!
I wanted to drop a quick question. I've installed it on a windows laptop and it works perfectly fine. But when I use your Mac installation files, the process goes smoothly, with no errors, but in the end the "Batch image manipulation" option just doesn't appear in the File menu. I'm using an M1 Mac, so maybe the problem lies there. 
Do you have any idea?
I would be tremendously greatful if you could help.

**Comment by jeinselen** [comment]
Same issue on an Intel MacBook Pro; installation has no errors, I allow Bimp to open from the system security settings, but no matter how many times I restart Gimp, it never appears in the File menu.

Versions:

- Gimp 2.10.38 (from official download)
- Bimp 2.6 (from here on GitHub)
- macOS Sonoma 14.4.1

**Comment by raphillip** [comment]
I have a similar issue. MacOS Ventura 13.1. Latest GIMP. I installed BIMP a few days ago and it worked fine. Today, it said there was an error. I tried to reinstall it. It doesn't appear under the File menu.
I deleted the BIMP folder from /Users/rphillip/Library/Application Support/GIMP/2.10 and tried again. 
This is what Terminal reported:
Last login: Wed May 15 10:15:39 on console
/Applications/gimp-plugin-bimp_macos/install ; exit;
rphillip@MacBook-Air ~ % /Applications/gimp-plugin-bimp_macos/install ; exit;
mkdir: /Users/rphillip/Library/Application Support/GIMP/2.10/plug-ins: File exists
BIMP is installed. You can close the program.

Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

[Process completed]

I went to Sytem Settings and allowed the download. I got
macOS cannot verify the developer of “bimp”. Are you sure you want to open it?
and clicked OK.
Restarted GIMP. Nothing appears under the file menu...
Confused, Rob

**Comment by TheodorEmanuelsson** [comment]
I was trying to install BIMP on an ARM macbook but failed unfortunately. I think that it should work for x_86 macbooks, so I'll still share my approach. Running the bimp executable in `Users/<NAME>/Library/Application Support/GIMP/2.10/plug-ins/bimb/`, I got the error that `/Applications/GIMP-2.10.app/Contents/Resources/lib/libgimpui-2.0.0.dylib` doesn't exist. My application was however named `GIMP.app`, so I renamed it to `GIMP-2.10.app` instead. Now when trying to run the `bimb` executable, I get that it find the file, but `incompatible architecture (have 'arm64', need 'x_86_64)`

**Comment by maxoakland** [comment]
I'm also having this problem. Batch Image Manipulation just won't show up in the file menu

**Comment by maxoakland** [comment]
> I was trying to install BIMP on an ARM macbook but failed unfortunately. I think that it should work for x_86 macbooks, so I'll still share my approach. Running the bimp executable in `Users/<NAME>/Library/Application Support/GIMP/2.10/plug-ins/bimb/`, I got the error that `/Applications/GIMP-2.10.app/Contents/Resources/lib/libgimpui-2.0.0.dylib` doesn't exist. My application was however named `GIMP.app`, so I renamed it to `GIMP-2.10.app` instead. Now when trying to run the `bimb` executable, I get that it find the file, but `incompatible architecture (have 'arm64', need 'x_86_64)`

That last error is saying you installed the M1 version on an Intel Mac

**Comment by egarpor** [comment]
Same issues on macOS Sonoma 14.5 on a Macbook Pro M1 and GIMP 2.10.32. Followed the steps of the readme.txt ("Allow Anyway" to unlock the BIMP executable, etc), but the plugin does not show up in the "File" menu.

**Comment by egarpor** [comment]
In case it is helpful, as pointed out in [this Reddit thread](https://www.reddit.com/r/GIMP/comments/1262pw4/bimp_doesnt_do_anything_on_mac_os/), a workaround is to downgrade to [version 2.10.22](https://download.gimp.org/gimp/v2.10/osx/), and then the menu option appear and is operational. I just tested it with macOS Sonoma 14.6.1 on an Apple M1 processor. Specifically, I downloaded [gimp-2.10.22-x86_64-3.dmg](https://download.gimp.org/gimp/v2.10/osx/gimp-2.10.22-x86_64-3.dmg), which works fine.

**Comment by TheodorEmanuelsson** [comment]
> > I was trying to install BIMP on an ARM macbook but failed unfortunately. I think that it should work for x_86 macbooks, so I'll still share my approach. Running the bimp executable in `Users/<NAME>/Library/Application Support/GIMP/2.10/plug-ins/bimb/`, I got the error that `/Applications/GIMP-2.10.app/Contents/Resources/lib/libgimpui-2.0.0.dylib` doesn't exist. My application was however named `GIMP.app`, so I renamed it to `GIMP-2.10.app` instead. Now when trying to run the `bimb` executable, I get that it find the file, but `incompatible architecture (have 'arm64', need 'x_86_64)`
> 
> That last error is saying you installed the M1 version on an Intel Mac

No, the last errors says that I have arm64 (Apple Silicon architecture cpu) but need an x_86 (old intel chips for mac).

**Comment by jackie-maxx** [comment]
> In case it is helpful, as pointed out in [this Reddit thread](https://www.reddit.com/r/GIMP/comments/1262pw4/bimp_doesnt_do_anything_on_mac_os/), a workaround is to downgrade to [version 2.10.22](https://download.gimp.org/gimp/v2.10/osx/), and then the menu option appear and is operational. I just tested it with macOS Sonoma 14.6.1 on an Apple M1 processor. Specifically, I downloaded [gimp-2.10.22-x86_64-3.dmg](https://download.gimp.org/gimp/v2.10/osx/gimp-2.10.22-x86_64-3.dmg), which works fine.

thank you it work for me on mac os Sequoia on chip M3
